src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -201,6 +205,14 @@ grpc::Status MasterMetadataServiceImpl::HandleFileChunkRead(
 
   // Set the file chunk metadata in reply
   *reply->mutable_metadata() = file_chunk_metadata_or.ValueOrDie();
+  // Set the location information, as we always inquiry the chunk server 
+  // manager instead of caching on the master
+  auto chunk_server_locations(
+      chunk_server_manager().GetChunkLocations(chunk_handle));
```

Comment:
GetChunkLocations can return an empty set if all the servers storing that chunk are down. We should check return and handle that scenario?